--- a/arch/arm/dts/Makefile	2022-12-22 18:27:05.132319631 +0900
+++ b/arch/arm/dts/Makefile	2022-12-22 18:27:25.911588845 +0900
@@ -586,7 +586,8 @@
 	sun7i-a20-primo73.dtb \
 	sun7i-a20-wexler-tab7200.dtb \
 	sun7i-a20-wits-pro-a20-dkt.dtb \
-	sun7i-a20-yones-toptech-bd1078.dtb
+	sun7i-a20-yones-toptech-bd1078.dtb \
+	sun7i-a20-marsboard.dtb
 dtb-$(CONFIG_MACH_SUN8I_A23) += \
 	sun8i-a23-evb.dtb \
 	sun8i-a23-gt90h-v4.dtb \
